secrets.go

```diff
@@ -0,0 +1,106 @@
+package clues
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+)
+
+type hashAlg int
+
+const (
+	SHA256 hashAlg = iota
+	HMAC_SHA256
+	Plaintext
+)
+
+var hashingAlgorithm = HMAC_SHA256
+
+type Concealer interface {
+	Conceal() string
+}
+
+type secret struct {
+	s string
+	v any
+	h string
+}
+
+func (s secret) String() string  { return s.s }
+func (s secret) Conceal() string { return s.h }
+
+// Mask embeds the value in a clues. secret where the
+// Conceal() call always returns a flat string: "***"
+func Mask(a any) secret {
+	str := marshal(a)
+
+	return secret{
+		s: str,
+		v: a,
+		h: "***",
+	}
+}
+
+// Hide embeds the value in a clues. secret where the
+// Conceal() call contains a truncated hash of value.
+// The hash function defaults to SHA256, but can be
+// changed through configuration.
+func Hide(a any) secret {
+	str := marshal(a)
+
+	return secret{
+		s: str,
+		v: a,
+		h: Conceal(hashingAlgorithm, str),
+	}
+}
+
+// HideAll is a quality-of-life wrapper for transforming
+// multiple values to clues.secrete structs.
```

Comment:
can't tell if this is ye olde English or you're just oozing with enthusiasm for this PR
```suggestion
// multiple values to clues.secret structs.
```